frontend/src/model/beta/Releases.tsx

```diff
@@ -1,18 +1,24 @@
-import { Box, Button, Stack } from '@mui/material'
+import { Box, Button, Stack, Typography } from '@mui/material'
 import { useEffect, useState } from 'react'
+import EmptyBlob from 'src/common/EmptyBlob'
 
+import { useGetReleasesForModelId } from '../../../actions/release'
+import { ModelInterface } from '../../../types/types'
 import DraftNewReleaseDialog from '../DraftNewReleaseDialog'
 import ModelReleaseDisplay from '../ModelReleaseDisplay'
 
-export default function Releases({ model }: { model: any }) {
+export default function Releases({ model }: { model: ModelInterface }) {
   const [latestRelease, setLatestRelease] = useState<string>('')
   const [openDraftNewRelease, setOpenDraftNewRelease] = useState(false)
 
+  const { releases, isReleasesLoading } = useGetReleasesForModelId(model.id)
+
   useEffect(() => {
-    if (model && model.releases) {
-      setLatestRelease(model.releases.sort((a, b) => a.semver < b.semver)[0].semver)
+    if (model && releases) {
+      const sortedReleases = releases.sort((a, b) => (a.semver < b.semver ? 1 : b.semver > a.semver ? -1 : 0))
```

Comment:
This sort is being performed multiple times in this component. Consider using `useMemo` to ensure we're only doing it when needed.
```
const sortedReleases = useMemo(() => releases.sort(sortByReleaseVersionDescending), [releases])
```

Reply:
Unresolved, still outstanding